src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Tests/ExceptionToUnmanagedMarshallerTests.cs

```diff
@@ -0,0 +1,54 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Marshalling;
+using Xunit;
+
+namespace ComInterfaceGenerator.Tests
+{
+    public unsafe partial class ExceptionToUnmanagedMarshallerTests
+    {
+        [CustomMarshaller(typeof(Exception), MarshalMode.UnmanagedToManagedOut, typeof(CustomExceptionAsHResultMarshaller))]
+        public static class CustomExceptionAsHResultMarshaller
+        {
+            public static int LastException { get; private set; }
+
+            public static int ConvertToUnmanaged(Exception e)
+            {
+                return LastException = ExceptionAsHResultMarshaller<int>.ConvertToUnmanaged(e);
+            }
+        }
+
+        [GeneratedComInterface(ExceptionToUnmanagedMarshaller = typeof(CustomExceptionAsHResultMarshaller))]
+        [Guid("90F3657D-23B7-44C4-85DD-80BD1F5266E7")]
+        public partial interface ICustomExceptionMarshallerComInterface
+        {
+            void ThrowException();
+        }
+
+        [GeneratedComClass]
+        public partial class CustomExceptionMarshallerComClass : ICustomExceptionMarshallerComInterface
+        {
+            public void ThrowException() => throw new NotImplementedException();
+        }
+
+        [Fact]
+        public void TestCustomExceptionMarshaller()
+        {
+            CustomExceptionMarshallerComClass comObject = new();
+            StrategyBasedComWrappers wrappers = new();
+            nint nativeUnknown = wrappers.GetOrCreateComInterfaceForObject(comObject, CreateComInterfaceFlags.None);
+            Marshal.QueryInterface(nativeUnknown, typeof(ICustomExceptionMarshallerComInterface).GUID, out nint nativeInterface);
```

Comment:
This isn't unreasonable but not our normal process here. What I would do is create a wrapping RCW for the CCW, just call `GetOrCreateObjectForComInterface()`, don't unwrap it. This will create the following structure we can test:

`CustomExceptionMarshallerComClass` -> `CCW` -> `RCW` -> `ICustomExceptionMarshallerComInterface`

At this point you can use the `ICustomExceptionMarshallerComInterface` and catch the expected exception and will verify the full stack. It will also handle the potential leaking that is occuring for the two `Marshal.Release()` calls and exposing the details of the vtable layout.